### PR TITLE
add programmes sort functionality

### PR DESCRIPTION
**Pull request by ancazapuc9**
Added sort functionality for programmes.
Pls review and give me feedback.

### CODE REVIEW

File: src/test/java/com/metabroadcast/atlas/glycerin/GlycerinIntegrationTest.java [hunk 1]
```diff
@@ -166,4 +160,71 @@ public void testProgrammesQueryWithContributions() throws GlycerinException {
...
+                        withAvailability(AvailabilityOption.AVAILABLE).
+                        withMixins(ProgrammesMixin.AVAILABLE_VERSIONS).
+                        withPageSize(25).
+                        sortBy(ProgrammesSort.SCHEDULED_START, ProgrammesSortDirection.DESCENDING).
+                        build();
+        GlycerinResponse<Programme> response = glycerin.execute(query);
+
+        List<Long> startSchedules = new ArrayList<>();
+        for (Programme programme : response.getResults()) {
+            if (programme.isSeries() || programme.isClip()) continue;
```
Comment (emilssolmanis) [inline src/test/java/com/metabroadcast/atlas/glycerin/GlycerinIntegrationTest.java]: We don't do 1-line `if`s either. Braces for everything.

File: src/test/java/com/metabroadcast/atlas/glycerin/GlycerinIntegrationTest.java [hunk 2]
```diff
@@ -166,4 +160,71 @@ public void testProgrammesQueryWithContributions() throws GlycerinException {
...
+            AvailableVersions availableVersions = programme.getAsEpisode().getAvailableVersions();
+            Long scheduledStartTime = obtainScheduledStartTime(availableVersions);
+            if (scheduledStartTime == null) continue;
```
Comment (emilssolmanis) [inline src/test/java/com/metabroadcast/atlas/glycerin/GlycerinIntegrationTest.java]: Same.

File: src/queries/java/com/metabroadcast/atlas/glycerin/generator/FeedQueryGenerator.java [hunk 3]
```diff
@@ -2,22 +2,18 @@
 
 import static com.google.common.base.Preconditions.checkState;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
+import com.metabroadcast.atlas.glycerin.model.*;
```
Comment (emilssolmanis) [inline src/queries/java/com/metabroadcast/atlas/glycerin/generator/FeedQueryGenerator.java]: See other comment about imports.

File: src/queries/java/com/metabroadcast/atlas/glycerin/generator/FeedQueryGenerator.java [hunk 4]
```diff
@@ -77,53 +73,71 @@ public void generateQuery(Feed feed) {
             if (transformedType == null) {
                 return;
             }
-            
-            JDefinedClass cls = pkg._class(JMod.PUBLIC|JMod.FINAL, String.format("%sQuery", feed.getName()));
+
+            JDefinedClass cls = pkg._class(JMod.PUBLIC | JMod.FINAL, String.format("%sQuery", feed.getName()));
```
Comment (emilssolmanis) [inline src/queries/java/com/metabroadcast/atlas/glycerin/generator/FeedQueryGenerator.java]: All of these whitespace changes create a lot of unnecessary noise and make it much harder to review the actual functional changes.

The code is dense as is, you need to either
1. make the whitespace fixup a separate commit so we can review just the functional stuff
or
2. just get rid of the whitespace modifications altogether

**Comment by ancazapuc9** [comment]
Changes done.
If still not OK with whitespace / spaces/ tabs I mean can u pls provide some code style config file ?
I am using IntelijIdea 2016.2 and I am using their default code style.

**Comment by emilssolmanis** [comment]
@ancazapuc9 Merged it, but we get a test failure -- it seems the descending sort by `scheduled_start` doesn't work. Does it actually pass for you? I might have to revert this otherwise.

**Comment by ancazapuc9** [comment]
Hello,

For me it worked, I will retest 
Anca

**Comment by ancazapuc9** [comment]
Found the pb, u can review it
